test/Tests/Writers/AsciiDoc.hs

```diff
@@ -90,11 +87,6 @@ tests = [ testGroup "emphasis"
                                            [ "* [ ] a"
                                            , "* [X] b"
                                            ]
-          , test (asciidocWithOpts def) "bullet without task_lists" $
-               bulletList [plain "☐ a", plain "☒ b"] =?> unlines
-                                           [ "* ☐ a"
-                                           , "* ☒ b"
-                                           ]
```

Comment:
If we leave `task_lists` as an extension that can be disabled, then it makes sense to test what happens when it is disabled.  But I'm still not sure why we should make task list rendering something you *can* disable with an extension.

So, I'd favor

- removing `task_list` from the available extensions for asciidoc.
- changing the code to do the task list transformation if asciidoctorVariant is set in WriterState, otherwise not.  (A cursory glance through the original asciidoc user guide https://asciidoc-py.github.io/userguide.html doesn't show anything about checklists, so it seems to be an asciidoctor-only feature.)

With these changes, task list syntax will be used automatically with `-t asciidoctor` but not with `-t asciidoc`, which I think is what we want.  Extensions should be used only when it makes sense to disable them.

Reply:
Fixed, review please